Source/Diagnostics/FullDiagnostics.cpp

```diff
@@ -177,145 +177,138 @@ FullDiagnostics::InitializeFieldFunctorsRZopenPMD (int lev)
             warpx.get_pointer_current_fp(lev, dim)->nComp() == ncomp_multimodefab );
     }
 
-    amrex::Vector< std::string > temp_m_varnames(m_varnames);
-    // Clear any pre-existing vector to release stored data.
-    m_varnames.clear();
-    m_all_field_functors[lev].clear();
-
     // Species index to loop over species that dump rho per species
     int i = 0;
-    bool convert2cartesian;
-    int ncomp;
-    if (m_format == "openpmd") {
```

Comment:
This function is actually only called when `m_format == "openpmd"` (see the actual name of the function: `InitializeFieldFunctorsRZopenPMD`). Therefore, the `else` part of this `if` condition is never used.

Reply:
As a consequence, we also explicitly replace `convert2cartesian` by `false` is the rest of the function.